server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobQueues.java

```diff
@@ -51,6 +52,46 @@ public void add(TabletMetadata tabletMetadata, Collection<CompactionJob> jobs) {
     }
   }
 
+  public KeySetView<CompactionExecutorId,CompactionJobPriorityQueue> getQueueIds() {
+    return priorityQueues.keySet();
+  }
+
+  public long getQueueMaxSize(CompactionExecutorId executorId) {
+    // Handle if the queue no longer exists.
+    if (priorityQueues.get(executorId) == null) {
+      return 0;
+    }
+    return priorityQueues.get(executorId).getMaxSize();
```

Comment:
There is a race condition calling get() twice on the map.  Could do the following to only call it once an avoid the race.

```suggestion
    var prioQ = priorityQueues.get(executorId);
    return prioQ == null ? 0 : prioQ.getMaxSize();
```